scss/_list-group.scss

```diff
@@ -56,7 +56,7 @@
   // Hover state
   &:hover,
   &:focus {
-    z-index: 1; // Place hover/focus items above their siblings for proper border styling
+    z-index: 2; // Place hover/focus items above their siblings for proper border styling
```

Comment:
Why do we need to increment this value? I haven't found the broken use case.

Reply:
I think we don't, I think this is leftover from some explorations on the colors from earlier. List group is the only instance I can see where we remove a border instead of overlapping them like in button groups and input groups.